addons/website/models/ir_ui_view.py

```diff
@@ -364,10 +363,10 @@ def _handle_visibility(self, do_raise=True):
         if self.visibility and not request.env.user.has_group('website.group_website_designer'):
             if (self.visibility == 'connected' and request.website.is_public_user()):
                 error = werkzeug.exceptions.Forbidden()
-            elif self.visibility == 'restricted_group' and self.visibility_group:
+            elif self.visibility == 'restricted_group' and self.groups_id:
                 # special case, to avoid employee.user_ids
-                if (self.visibility_group.get_external_id() == 'base.group_user' and not request.env.user.share) or \
-                        request.env.user.id not in self.visibility_group.sudo().users.ids:
+                if (self.groups_id.get_external_id() == 'base.group_user' and not request.env.user.share) or \
```

Comment:
`groups_id` being a `m2m`, does it works when the view has multiple groups set?

Reply:
No it does not you are right, but I do not think it worked before either as `get_external_id` returns a map. @JKE-be Can I push on this or I let you handle it ?

Reply:
it is an old optimization no mroe used since the first if will handle the case 'signed in'
But yes we could remove this leftover code, but feature is working as expected with or without it.